### PR TITLE
Add support for using http to connect S3

### PR DESCRIPTION
**Pull request by lss602726449**
Currently, We always use HTTPS to connect to S3. We should also support HTTP to do so.

### CODE REVIEW

File: sql/pg_duckdb--0.0.1.sql [hunk 1]
```diff
@@ -117,6 +117,7 @@ CREATE TABLE secrets (
     region TEXT,
     endpoint TEXT,
     r2_account_id TEXT,
+    use_ssl BOOLEAN,
```
Comment (JelteF) [inline sql/pg_duckdb--0.0.1.sql:120]: nitpick, seems nicer to set a default here:
```suggestion
    use_ssl BOOLEAN DEFAULT true,
```

Comment (lss602726449) [reply sql/pg_duckdb--0.0.1.sql:120]: Good idea, Thanks for your suggestion.

File: src/pgduckdb_duckdb.cpp [hunk 2]
```diff
@@ -117,6 +117,9 @@ DuckdbCreateConnection(List *rtables, PlannerInfo *planner_info, List *needed_co
 		if (is_r2_cloud_secret) {
 			appendStringInfo(secret_key, ", ACCOUNT_ID '%s'", secret.endpoint.c_str());
 		}
+		if (!secret.use_ssl) {
+			appendStringInfo(secret_key, ", USE_SSL '%s'", "FALSE");
```
Comment (wuputah) [inline src/pgduckdb_duckdb.cpp:121]: No reason to format here:

```suggestion
			appendStringInfo(secret_key, ", USE_SSL 'FALSE'");
```

**Review by wuputah — COMMENTED** [review]
I'm a little surprised DuckDB supports this. It seems very dangerous to me. What's the use case here? The default is `true` so that's good, and the logic is simple enough, I'm just worried about some future bug.

We should really get some kind of test coverage here, though I'm not sure how we want to do that if it's actually going to S3. Could we somehow test that the secret is created correctly in DuckDB?

**Comment by lss602726449** [comment]
I just use a local minio without HTTPS. So I suppose to add this feature. This is support in duckdb in the readme https://github.com/duckdb/duckdb/blob/862852fa395b99735e5713cb55d0cea1d9320659/test/sql/copy/s3/README.md?plain=1#L46

**Review by wuputah — APPROVED** [review]

**Pull request merged**